src/operations/extrusions/extrudeFromSlices.js

```diff
@@ -0,0 +1,91 @@
+const { mat4 } = require('../../math')
+
+const { geom2, geom3, poly3 } = require('../../geometry')
+
+const slice = require('./slice')
+
+const extrudeWalls = require('./extrudeWalls')
+
+// NOTE: function definition is required in order to access 'this'
+function defaultCallback (t, index) {
+  let baseSlice = null
+  if (geom2.isA(this)) baseSlice = slice.fromSides(geom2.toSides(this))
+  if (poly3.isA(this)) baseSlice = slice.fromPoints(poly3.toPoints(this))
+
+  return t === 0 || t === 1 ? slice.transform(mat4.fromTranslation([0, 0, t]), baseSlice) : null
+}
+
+/**
+ * Extrude a solid from the slices as returned by the callback function.
+ *
+ * @param {Object} options - options for extrude
+ * @param {Integer} [options.numberOfSlices=2] the number of slices to be generated by the callback
+ * @param {Boolean} [options.isCapped=true] the solid should have caps at both start and end
+ * @param {Function} [options.callback] the callback function that generates each slice
+ * @param {Object} base - the base object which is used to create slices (see below for callback information)
+ * @return {geom3} the extruded shape
+ *
+ * @example
+ * let newslice = callback(t, index)
+ *     this : the given base object
+ *     arguments: t = [0..1], index = [0..numberOfSlices - 1]
+ *     return: slice or null (to skip)
+ */
+const extrudeFromSlices = (options, base) => {
+  const defaults = {
+    numberOfSlices: 2,
+    isCapped: true,
+    callback: defaultCallback
+  }
+  let { numberOfSlices, isCapped, callback } = Object.assign({ }, defaults, options)
+
+  if (numberOfSlices < 2) throw new Error('numberOfSlices must be 2 or more')
+
+  const sMax = numberOfSlices - 1
+
+  let startSlice = null
+  let endSlice = null
+  let prevSlice = null
+  let polygons = []
+  for (let s = 0; s < numberOfSlices; s++) {
+    // invoke the callback function to get the next slice
+    // NOTE: the base object becomes 'this' inside the callback
+    // NOTE: callback can return null to skip the slice
+    let currentSlice = callback.call(base, s / sMax, s)
```

Comment:
Nice! I like it. It will take a little time to re-work. How does this look for the callback signiture?

```
let newslice = callback(base, t, index)
/*    base : the base geometry as given to extrudeFromSlices
 *    t = [0..1]
 *    index = [0..numberOfSlices - 1]
 *    return: slice or null (to skip)
 */
```

Reply:
What is the 't' parameter? Percentage complete? Should this be kept? It seems redundant.

Reply:
@kaosat-dev thoughts?

Reply:
loved what you did :) Groovy !